tests/sources/test_salesforce.py

```diff
@@ -1121,6 +1555,138 @@ async def test_get_all_with_content_docs_and_extraction_service(mock_responses):
             TestCase().assertCountEqual(content_document_records, [expected_doc])
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "soql_query, modified_query",
+    [
+        ("SELECT Id, Name FROM Account", "SELECT Id, Name FROM Account"),
+        (
+            "SELECT FIELDS(ALL) FROM Account",
+            "SELECT FIELDS(ALL) FROM Account LIMIT 200 OFFSET 0",
+        ),
+        (
+            "SELECT FIELDS(CUSTOM) FROM Account LIMIT 200",
+            "SELECT FIELDS(CUSTOM) FROM Account LIMIT 200 OFFSET 0",
+        ),
+        (
+            "SELECT FIELDS(ALL) FROM Account LIMIT 200 OFFSET 0",
+            "SELECT FIELDS(ALL) FROM Account LIMIT 200 OFFSET 0",
+        ),
+    ],
+)
+async def test_modify_query(soql_query, modified_query):
```

Comment:
Can you add a case to check the logic around adding the `LastModifiedDate` clause?